### PR TITLE
[WIP] mesos: 1.4.1 -> 1.10.0

### PR DESCRIPTION
**Pull request by ludovicc**
###### Motivation for this change

Upgrade Mesos from 1.4.1 to 1.10.0, fix this package broken since Dec 2019 and a dependency of Spark.

Need help to complete the packaging.

###### Things done

- refreshed nixos patch
- currently autoconfigure checks failing

```
checking google/protobuf/map.h usability... yes
checking google/protobuf/map.h presence... yes
checking for google/protobuf/map.h... yes
checking for /nix/store/ayiyc1a33cddxbgvrjhqh152inncbcn5-protobuf-3.6.1.3/include/google/protobuf/wrappers.proto... yes
checking for protoc... protoc
checking python2 module: google.protobuf... no
configure: error: failed to find required module google.protobuf
```

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ludovicc** [comment]
@cstrahan can you help me to complete this upgrade? Thanks

File: pkgs/applications/networking/cluster/mesos/default.nix [hunk 1]
```diff
@@ -25,23 +25,23 @@ let
...
   patches = [
     # https://reviews.apache.org/r/36610/
     # TODO: is this still needed?
     ./rb36610.patch
 
-    # see https://github.com/cstrahan/mesos/tree/nixos-${version}
+    # see https://github.com/ludovicc/mesos/tree/nixos-${version}
     ./nixos.patch
   ];
   nativeBuildInputs = [
```
Comment (jonringer) [inline pkgs/applications/networking/cluster/mesos/default.nix:47]: to solve your immediate issue
```suggestion
  nativeBuildInputs = [
    python2.pkgs.protobuf
```

**Comment by flokli** [comment]
mesos, chronos and marathon have been removed in https://github.com/NixOS/nixpkgs/pull/95507, so this PR needs to be rebased onto master. Please also re-introduce the tests (in python) while doing so.

File: pkgs/applications/networking/cluster/mesos/nixos.patch [hunk 2]
```diff
@@ -543,7 +473,11 @@ index 57d4ccd..68c9577 100644
           << " protocol ip"
           << " prio " << Priority(IP_FILTER_PRIORITY, NORMAL).get() << " u32"
           << " flowid ffff:0"
-@@ -4011,7 +4013,7 @@ string PortMappingIsolatorProcess::scripts(Info* info)
+          << " match ip dst "
+-         << net::IP::Network::LOOPBACK_V4().address()
++         << net::IPNetwork::LOOPBACK_V4().address()
```
Comment (volth) [inline pkgs/applications/networking/cluster/mesos/nixos.patch:478]: No!!!! It breaks the build

File: pkgs/applications/networking/cluster/mesos/nixos.patch [hunk 3]
```diff
@@ -561,7 +495,12 @@ index 57d4ccd..68c9577 100644
             << " protocol ip"
             << " prio " << Priority(IP_FILTER_PRIORITY, NORMAL).get() << " u32"
             << " flowid ffff:0"
-@@ -4032,14 +4034,14 @@ string PortMappingIsolatorProcess::scripts(Info* info)
+            << " match ip dst "
+-           << net::IP::Network::LOOPBACK_V4().address()
++           << net::IPNetwork::LOOPBACK_V4().address()
```
Comment (volth) [inline pkgs/applications/networking/cluster/mesos/nixos.patch:500]: No!!

File: pkgs/applications/networking/cluster/mesos/nixos.patch [hunk 4]
```diff
@@ -578,8 +517,10 @@ index 57d4ccd..68c9577 100644
           << " protocol ip"
           << " prio " << Priority(ICMP_FILTER_PRIORITY, NORMAL).get() << " u32"
           << " flowid ffff:0"
-@@ -4048,9 +4050,9 @@ string PortMappingIsolatorProcess::scripts(Info* info)
-          << net::IP::Network::LOOPBACK_V4().address() << "\n";
+          << " match ip protocol 1 0xff"
+          << " match ip dst "
+-         << net::IP::Network::LOOPBACK_V4().address() << "\n";
++         << net::IPNetwork::LOOPBACK_V4().address() << "\n";
```
Comment (volth) [inline pkgs/applications/networking/cluster/mesos/nixos.patch:523]: here too

**Comment by flokli** [comment]
What's the status of this? Can this be rebased to latest master, as written in https://github.com/NixOS/nixpkgs/pull/95631#issuecomment-675085661?

**Pull request closed without merging**